server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -466,9 +468,11 @@ public boolean equals(Object obj) {
             if (getClass() != obj.getClass())
                 return false;
             Builder other = (Builder) obj;
-            if (!Objects.equals(aggregationBuilders, other.aggregationBuilders))
+
+            // compare aggregations independent of their order
+            if (!aggregationBuilders.containsAll(other.aggregationBuilders))
```

Comment:
I don't think this will satisfy the "equals" contract: !(A, B).constainsAll(A, B, C) but (A, B, C).constainsAll(A, B). I think and additional size check will solve this though.